drivers/soundwire/intel_init.c

```diff
@@ -263,6 +269,25 @@ static struct sdw_intel_ctx
 		md->driver->probe(md, link);
 
 		list_add_tail(&link->list, &ctx->link_list);
+		bus = &link->cdns->bus;
```

Comment:
is `link->cdns` supposed to be assigned in above .probe()? Should we check for that to avoid a NULL dereference? Does that .probe() return an error code, can it fail?

Reply:
> is `link->cdns` supposed to be assigned in above .probe()? Should we check for that to avoid a NULL dereference? Does that .probe() return an error code, can it fail?

Indeed, it could be NULL if md->driver->probe fail. I guess we have to check the return value of md->driver->probe. Thanks.